symbolic-debuginfo/src/dwarf.rs

```diff
@@ -554,8 +556,23 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
                 | constants::DW_AT_start_scope => {
                     match self.inner.info.attr_ranges(self.inner.unit, attr.value())? {
                         Some(mut ranges) => {
-                            while let Some(range) = ranges.next()? {
-                                range_buf.push(range);
+                            while let Some(range) = match ranges.next() {
```

Comment:
Nit: The combination of `while let` and `match` is concise, but it honestly took me a bit to understand the flow (what is the block that opens immediately after the match block? Oh, it's the `while`'s body). This may just be me; it's fine to merge either way.

Reply:
you are absolutely right, it is confusing, but the `match` has to be in the `while let` header, we can’t move that out. And because of the early return, we also can’t move any of that into its own function :-(